### ISSUE TITLE
Proposal for variable routes

### ISSUE THREAD

**Issue by anddave** [jpsider/RestPS#67]
**Proposal for variable routes**
- As of RestPS 7.0.44 from PSGallery, only hardcoded routes are supported, which is to say that [Invoke-RequestRouter](https://github.com/jpsider/RestPS/blob/master/RestPS/private/Invoke-RequestRouter.ps1#L37) selects routes `where {$_.RequestURL -eq $RequestURL}`.  
- Consequently endpoints like "api/periodictable/{element}" are marginally feasible in small scale only and not feasible in large scale. 
- Proposal: we could add a regex match to Invoke-RequestRouter which would enable variable routes.
- For the sake of simplicity, I propose that: 
  - Map file should _only_ be changed to support regex matching
  - as opposed to adding any more notational load to the map file, let the endpoint scripts do _all_ the interpreting of variable routes beyond matching the regex.
- Some defensive code and unit testing may need to be added for mitigating the risk of accidental route doubling if a route is both an exact match _and_ a regex match.
- Branch and PR forthcoming soon
Thanks @jpsider great module

**Comment by jpsider** [comment]
Makes sense. I have not had the need to add this up to this point. But in dealing with other API's use this type of thing on a normal basis. I am looking forward to reviewing the PR!
